helios-client/src/main/java/com/spotify/helios/client/GoogleCredentialsAccessTokenProvider.java

```diff
@@ -74,12 +74,11 @@ static AccessToken getAccessToken(List<String> scopes) throws IOException {
       return null;
     }
 
-    if (!scopes.isEmpty()) {
-      credentials.createScoped(scopes);
-    }
-    credentials.refresh();
+    final GoogleCredentials newCredentials =
```

Comment:
Minor: `credentials` isn't `final` above, so you could just replace it inline.

```
if (!scopes.isEmpty()) {
    credentials = credentials.createScoped(scopes);
}
```

Reply:
+1 to this